Spotinst - adding event subscription module

jeffnoehren opened the pull request:
##### SUMMARY
Added new module for Spotinst Event Subscription 

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
Spotinst

##### ADDITIONAL INFORMATION
Added new module to interact with the Spotinst event subscription API

```paste below
result:
    type: str
    sample: sis-e62dfd0f
    returned: success
    description: Created Subscription successfully
```

Commit:
Spotinst - adding event subcription module

Spotinst - adding unit test for the spotinst_event_subscription module

Akasurde commented on lib/ansible/modules/cloud/spotinst/spotinst_event_subscription.py:
```diff
@@ -0,0 +1,281 @@
+    You will have to have a credentials file in this location - <home>/.spotinst/credentials
+    The credentials file must contain a row that looks like this
+    token = <YOUR TOKEN>
+    Full documentation available at https://help.spotinst.com/hc/en-us/articles/115003530285-Ansible-
```
```suggestion
    Full documentation available at U(https://help.spotinst.com/hc/en-us/articles/115003530285-Ansible-)
```

jeffnoehren replied:
added

Akasurde commented on lib/ansible/modules/cloud/spotinst/spotinst_event_subscription.py:
```diff
@@ -0,0 +1,281 @@
+
+  id:
+    description:
+      - (String) Parameters used for Updating or Deleting subscription.
```
```suggestion
      - Parameters used for Updating or Deleting subscription.
    type: str
```

jeffnoehren replied:
added

Akasurde commented on lib/ansible/modules/cloud/spotinst/spotinst_event_subscription.py:
```diff
@@ -0,0 +1,281 @@
+def main():
+    fields = dict(
+        account_id=dict(type='str'),
+        token=dict(type='str'),
```
You can use `env_fallback` like 

```python
from ansible.module_utils.basic import env_fallback
...
token=dict(type='str', fallback=(env_fallback, ['SPOTINST_TOKEN'])),

jeffnoehren replied:
added

Akasurde commented on lib/ansible/modules/cloud/spotinst/spotinst_event_subscription.py:
```diff
@@ -0,0 +1,281 @@
+
+def main():
+    fields = dict(
+        account_id=dict(type='str'),
```
```suggestion
        account_id=dict(type='str', fallback=(env_fallback, ['SPOTINST_ACCOUNT_ID', 'ACCOUNT']),
```

jeffnoehren replied:
added

Akasurde commented on lib/ansible/modules/cloud/spotinst/spotinst_event_subscription.py:
```diff
@@ -0,0 +1,281 @@
+
+    account = module.params.get('account_id')
+    if not account:
+        account = os.environ.get('SPOTINST_ACCOUNT_ID') or os.environ.get('ACCOUNT')
```
No need for this if you are using `env_fallback` mentioned in below comment.

jeffnoehren replied:
added in!

Akasurde reviewed (changes requested):
Could you please add a unit test for this change ? Thanks.

jeffnoehren commented:
@Akasurde Hey I added in the changes you suggested as well as a unit test for this module! Also I have three other PR's open right now that I went through and updated the same changes and added unit tests for those three other modules. Please let me know if there is anything else that is required to move forward! 

https://github.com/ansible/ansible/pull/51517
https://github.com/ansible/ansible/pull/51516
https://github.com/ansible/ansible/pull/51514

s-hertel commented on changelogs/fragments/spotinst-add-event-subscription-module.yaml:
```diff
@@ -0,0 +1,2 @@
+minor_changes:
```
New modules don't need a changelog fragment as they're automatically generated.

Akasurde commented on lib/ansible/modules/cloud/spotinst/spotinst_event_subscription.py:
```diff
@@ -0,0 +1,287 @@
+
+  protocol:
+    description:
+      - (String) Type of desired protocol
```
```suggestion
      - Type of desired protocol
    type: str
```

Akasurde commented on lib/ansible/modules/cloud/spotinst/spotinst_event_subscription.py:
```diff
@@ -0,0 +1,287 @@
+EXAMPLES = """
+#In this basic example, we create an event subscription
+
+- hosts: localhost
```
Replace line 81 to 97 - 

with 
```yaml
    - name: Create an event subscription
      spotinst_event_subscription:
        account_id:
        token:
        state: present
        id: sis-e62dfd0f
        resource_id: sig-992a78db
        protocol: web
        endpoint: https://webhook.com
        event_type: GROUP_UPDATED
        event_format: { "subject" : "%s", "message" : "%s" }
      register: result
```

Akasurde commented:
Ping me once you are done with the change. I will take a look into other PRs you mentioned.
